text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+    </td>
+   </tr>
+   <tr>
+    <td>Host
+    </td>
+    <td>"host"
+    </td>
+    <td>host.id
+    </td>
+    <td>host.name
+<p>
+host.type
+<p>
+host.image.id
+<p>
+host.image.name
+    </td>
+   </tr>
+   <tr>
+    <td>K8s Pod
+    </td>
+    <td>"k8s.pod"
+    </td>
+    <td>k8s.pod.uid (required)
+<p>
+k8s.cluster.name
+    </td>
+    <td>Any pod labels
+    </td>
+   </tr>
+   <tr>
+    <td>K8s Pod Container
+    </td>
+    <td>"container"
+    </td>
+    <td>k8s.pod.uid (required)
+<p>
+k8s.cluster.name
+<p>
+container.name
+    </td>
+    <td>Any container labels
+    </td>
+   </tr>
+</table>
+
+See more examples showing nuances of Id field composition in the
+[Entity Identification](#entity-identification) section.
+
+## Entity Events
+
+Information about Entities can be produced and communicated using 2
+types of entity events: EntityState and EntityDelete.
+
+### EntityState Event
+
+The EntityState event stores information about the _state_ of the entity
+at a particular moment of time. The data model of the EntityState event
+is the same as the entity data model with some extra fields:
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Timestamp
+    </td>
+    <td>nanoseconds
+    </td>
+    <td>The time since when the entity state is described by this event.
+The time is measured by the origin clock. The field is required.
+    </td>
+   </tr>
+   <tr>
+    <td>Interval
+    </td>
+    <td>milliseconds
+    </td>
+    <td>Defines the reporting period, i.e. how frequently the
+information about this entity is reported via EntityState events even if
+the entity does not change. The next expected EntityEvent for this
+entity is expected at (Timestamp+Interval) time. Can be used by
+receivers to infer that a no longer reported entity is gone, even if the
+EntityDelete event was not observed. Optional, if missing the interval
+is unknown.
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>
+    </td>
+    <td>See data model.
+
+</td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+   <tr>
+    <td>Attributes
+    </td>
+    <td>
+    </td>
+    <td>See data model
+</td>
+   </tr>
+</table>
+
+We say that an entity mutates (changes) when one or more of its
+descriptive attributes changes. A new descriptive attribute may be
+added, an existing descriptive attribute may be deleted or a value of an
+existing descriptive attribute may be changed. All these changes
+represent valid mutations of an entity over time. When these mutations
+happen the identity of the entity does not change.
+
+When the entity's state is changed it is expected that the source will
+emit a new EntityState event with a fresh timestamp and full list of
+values of all other fields.
+
+Entity event producers are recommended to periodically emit events even
```

Comment:
To prevent confusion about intent for this to be normative:

```suggestion
Entity event producers are expected to periodically emit events even
```

Reply:
Good catch. I think this really needs to be normative. Let me change it.

Reply:
Fixed.